src/Configuration/Option.php

```diff
@@ -157,4 +157,9 @@ final class Option
      * @var string
      */
     public const OPTION_NO_DIFFS = 'no-diffs';
+
+    /**
+     * @var string
+     */
+    public const TEMPLATE_TYPE = 'type';
```

Comment:
Value and name should be equal. That way we can avoid conflict with e.g. `ANY_TYPE = 'type';`

Reply:
So it should be template-type with or without prefix OPTION?

Reply:
What would you prefer?

Reply:
For the value template-type and for the constant name i would use TEMPLATE_TYPE also. Not using OPTION prefix here. But sometimes it is used. Do not see a pattern where and why.

Reply:
I see. The option prefix is for the command options. So we should add it here as well. Sorry too early and yesterday too late

Reply:
No troubles :)

Personally I would go for:

```php
const TEMPLATE_TYPE = 'template_type'
```

Reply:
Ok. But the pattern for options within a command is different. For existing ones it is always OPTION_FOO_BAR = 'option-with-dash'

Reply:
CLI is only one of possible inputs. The parameter can come from manual parameter too.

Reply:
But in cli mode i would prefer the dash. And this is visible for the user. In all other parts it is a hidden detail. Naming things is still hard and not solved by robots till now ;-)